### DISCUSSION TITLE
Failed to get Xref file keep all the time

### DISCUSSION THREAD

**Discussion by chkp-baselz** [oracle/opengrok#4464] [General]
Hello, 

I update modified src files daily, and run index daily, each day I get "failed to get Xref file" on different files(supported files of course), it keep failing all the time, some of the data file would have "*.org_opengrok", is there a way to fix this root issue, Am I doing something wrong? I index using this command 
`java     -Djava.util.logging.config.file=/opengrok/etc/logging.properties     -jar /opengrok/dist/lib/opengrok.jar     --ctags  /usr/local/bin/ctags     -s /opengrok/src -d /opengrok/data -H -P -S -G     -W /opengrok/etc/configuration.xml -U https://opengrok02:8443/source/`
it index and update the files, but the issue of "failed to get Xref file" keep happening. 

I also insert unsupported files to the src folder(the issue is about supported files extensions) may this cause the issue?

Thank you very much

**Comment by vladak** [comment]
Without seeing the indexer logs and doing more root causing (for how many files this fails, what is special about them, what they have in common, ...) it is hard to tell for sure however it looks like there might some bug with handling the renames of the `.org_opengrok` temporary files during reindex. There is a pre-existing issue filed for this problem I think - #3397.

1.6.9 contains the change for #3525 which essentially hides the problem in the UI.

**Comment by chkp-baselz** [comment]
@vladak log is attached, I didn't find anything but the unsupported files 
[indexLog.zip](https://github.com/oracle/opengrok/files/6296092/indexLog.zip)

**Comment by vladak** [comment]
The log is littered with `AccessDeniedException` for various index files under data root. This creates a lot of noise.

**Comment by chkp-baselz** [comment]
@vladak how could I fix it? is it the one causing all issue?

**Comment by vladak** [comment]
The data root needs to be writable for the indexer to succeed, you need to fix the file system permissions. It's hard to tell whether it could be causing the problem.

**Comment by vladak** [comment]
Out of curiosity: did only the data root have problem with permissions/ownership or also its sub-directories ?

**Comment by chkp-baselz** [comment]
@vladak the user I run index with have read/write permissions for all sub-directories of /opengrok/*
